### PR TITLE
Fixes #1500 - updated Bouncy Castle libs

### LINKED ISSUE #1500: Update Bouncy Castle libs
```
The ones we're currently using (1.46) are getting more out of date.
However its not just a case of updating the jars - some code changes are needed: http://www.bouncycastle.org/wiki/display/JA1/Porting+from+earlier+BC+releases+to+1.47+and+later

```

Original issue reported on code.google.com by `psiinon` on 2015-01-14 16:14:23

### PR DESCRIPTION
**Pull request by robocoder**

### CODE REVIEW

File: src/org/zaproxy/zap/extension/dynssl/SslCertificateUtils.java [hunk 1]
```diff
@@ -49,20 +49,21 @@
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/dynssl/SslCertificateUtils.java]: The import java.util.Vector should be removed.

**Comment by thc202** [comment]
It seems the update breaks the generation of the root CA cert:
33899 [AWT-EventQueue-0] ERROR org.zaproxy.zap.extension.dynssl.DynamicSSLPanel  - Error while generating Root CA certificate
java.lang.IllegalStateException: Errors during assembling root CA.
 at org.zaproxy.zap.extension.dynssl.SslCertificateUtils.createRootCA(SslCertificateUtils.java:136)
[...]
Caused by: java.lang.IllegalArgumentException: illegal object in getInstance: org.bouncycastle.asn1.DLSequence
 at org.bouncycastle.asn1.ASN1OctetString.getInstance(Unknown Source)
 at org.bouncycastle.asn1.ASN1OctetString.getInstance(Unknown Source)
 at org.bouncycastle.cert.jcajce.JcaX509ExtensionUtils.parseExtensionValue(Unknown Source)
 at org.zaproxy.zap.extension.dynssl.SslCertificateUtils.createRootCA(SslCertificateUtils.java:118)
 ... 87 more

(though this might be caused by the above) it's no longer possible to see it in the options:
27951 [AWT-EventQueue-0] ERROR org.zaproxy.zap.extension.dynssl.DynamicSSLPanel  - Error while extracting public part from generated Root CA certificate.
org.bouncycastle.util.io.pem.PemGenerationException: encoding exception: unknown object passed - can't encode.
	at org.bouncycastle.openssl.MiscPEMGenerator.generate(Unknown Source)
	at org.bouncycastle.util.io.pem.PemWriter.writeObject(Unknown Source)
	at org.zaproxy.zap.extension.dynssl.DynamicSSLPanel.setRootca(DynamicSSLPanel.java:236)

**Comment by robocoder** [comment]
Sorry, I reworked that patch this morning without testing it. =P  I'll look at it again tonight.

**Comment by thc202** [comment]
No problem at all! ;)

**Comment by robocoder** [comment]
Done. Thanks for the review.

**Comment by thc202** [comment]
Thank you!
How would you like to be credited?
https://github.com/zaproxy/zap-core-help/wiki/HelpCredits

**Comment by robocoder** [comment]
`Anthon Pang` is fine.

**Comment by thc202** [comment]
Done in zaproxy/zap-core-help@52d35b73d306ee5d47448b3f98c41d788521c9a3

**Pull request merged**